mono/utils/mono-threads.h

```diff
@@ -710,4 +719,39 @@ typedef void (*background_job_cb)(void);
 void mono_threads_schedule_background_job (background_job_cb cb);
 #endif
 
+typedef struct {
```

Comment:
I feel like this whole abstraction should probably be moved to a separate file, say, `mono-utility-thread.h`.

Reply:
Yeah, I can do that